src/data/cel.rs

```diff
@@ -63,6 +67,27 @@ impl Expression {
     }
 }
 
+fn url_decode(This(s): This<Arc<String>>) -> ResolveResult {
+    let mut map: HashMap<Key, Value> = HashMap::default();
+    for part in s.split('&') {
+        let mut kv = part.split('=');
+        if let (Some(key), Some(value)) = (kv.next(), kv.next()) {
+            match map.entry(decode(key).unwrap().into_owned().into()) {
+                Entry::Occupied(mut e) => {
+                    let v = e.get().clone();
```

Comment:
Need to check whether `v` is a `Value::List` here. Add to it if it is, otherwise `clone()` and replace the list as we do here.